src/bp-core/bp-core-functions.php

```diff
@@ -2484,49 +2484,69 @@ function bp_is_get_request() {
 /** Miscellaneous hooks *******************************************************/
 
 /**
- * Load the buddypress translation file for current language.
+ * Override translation file for current language.
  *
- * @since 1.0.2
+ * @since 14.0.0
  *
- * @see load_textdomain() for a description of return values.
- *
- * @return bool
+ * @param  string $file   Absolut path to the translation file to use.
+ * @param  string $domain The text domain to check against `buddypress`.
+ * @param  string $locale The current locale for the WordPress site.
+ * @return string Absolut path to the translation file to use.
  */
-function bp_core_load_buddypress_textdomain() {
```

Comment:
Do we have any unit tests for the old implementation?

Reply:
We don't have unit tests for old implementation. I've been looking into it to add some, but `WP_LANG_DIR` in WordPress Tests Suite can't be overridden, which means we need to rely on the dummy plugin WordPress uses, fake a translation and replicate the fact one or more of our strings are returned before `bp_core_loaded` is fired, in this case, `_load_textdomain_just_in_time()` will always set the location to the GlotPress one and `bp_core_load_buddypress_textdomain()` is useless.

I believe adding a unit test about it is not necessary.

Reply:
Sounds good.